process/block/preprocess/sovereignChainIncomingSCR.go

```diff
@@ -0,0 +1,191 @@
+		}
+
+		indexOfFirstTxToBeProcessed := pi.indexOfLastTxProcessed + 1
+		err = process.CheckIfIndexesAreOutOfBound(indexOfFirstTxToBeProcessed, pi.indexOfLastTxProcessedByProposer, miniBlock)
+		if err != nil {
+			return nil, err
+		}
+
+		for j := indexOfFirstTxToBeProcessed; j <= pi.indexOfLastTxProcessedByProposer; j++ {
+			if !haveTime() {
+				return nil, process.ErrTimeIsOut
+			}
+
+			txHash := miniBlock.TxHashes[j]
+			scr.scrForBlock.mutTxsForBlock.RLock()
+			txInfoFromMap, ok := scr.scrForBlock.txHashAndInfo[string(txHash)]
+			scr.scrForBlock.mutTxsForBlock.RUnlock()
+
+			if !ok || check.IfNil(txInfoFromMap.tx) {
+				log.Warn("missing transaction in ProcessBlockTransactions ", "type", miniBlock.Type, "txHash", txHash)
+				return nil, process.ErrMissingTransaction
+			}
+
+			currScr, ok := txInfoFromMap.tx.(*smartContractResult.SmartContractResult)
+			if !ok {
+				return nil, process.ErrWrongTypeAssertion
+			}
+
+			scr.saveAccountBalanceForAddress(currScr.GetRcvAddr())
+
+			_, err := scr.scrProcessor.ProcessSmartContractResult(currScr)
+			if err != nil {
+				return nil, err
+			}
+
+		}
+
+		createdMBs = append(createdMBs, miniBlock)
+	}
+
+	return createdMBs, nil
+}
+
+func (scr *sovereignChainIncomingSCR) ProcessMiniBlock(
+	miniBlock *block.MiniBlock,
+	haveTime func() bool,
+	_ func() bool,
+	_ bool,
+	partialMbExecutionMode bool,
+	indexOfLastTxProcessed int,
+	preProcessorExecutionInfoHandler process.PreProcessorExecutionInfoHandler,
+) ([][]byte, int, bool, error) {
+	if miniBlock.Type != block.SmartContractResultBlock {
+		return nil, indexOfLastTxProcessed, false, process.ErrWrongTypeInMiniBlock
+	}
+
+	numSCRsProcessed := 0
+	var err error
+	var txIndex int
+	processedTxHashes := make([][]byte, 0)
+
+	indexOfFirstTxToBeProcessed := indexOfLastTxProcessed + 1
+	err = process.CheckIfIndexesAreOutOfBound(int32(indexOfFirstTxToBeProcessed), int32(len(miniBlock.TxHashes))-1, miniBlock)
+	if err != nil {
+		return nil, indexOfLastTxProcessed, false, err
+	}
+
+	miniBlockScrs, miniBlockTxHashes, err := scr.getAllScrsFromMiniBlock(miniBlock, haveTime)
+	if err != nil {
+		return nil, indexOfLastTxProcessed, false, err
+	}
+
+	if scr.blockSizeComputation.IsMaxBlockSizeWithoutThrottleReached(1, len(miniBlock.TxHashes)) {
+		return nil, indexOfLastTxProcessed, false, process.ErrMaxBlockSizeReached
+	}
+
+	log.Debug("smartContractResults.ProcessMiniBlock: before processing")
+	defer func() {
+		log.Debug("smartContractResults.ProcessMiniBlock after processing")
+	}()
+
+	for txIndex = indexOfFirstTxToBeProcessed; txIndex < len(miniBlockScrs); txIndex++ {
+		if !haveTime() {
+			err = process.ErrTimeIsOut
+			break
+		}
+
+		if err != nil {
+			break
+		}
+
+		scr.saveAccountBalanceForAddress(miniBlockScrs[txIndex].GetRcvAddr())
+
+		snapshot := scr.handleProcessTransactionInit(preProcessorExecutionInfoHandler, miniBlockTxHashes[txIndex])
+		_, err = scr.scrProcessor.ProcessSmartContractResult(miniBlockScrs[txIndex])
```

Comment:
We have to go through the full consensus - but we do not need to execute the SCRs here. As far as I see from consensusV2 this processMiniBlock is called on create block. So we have to add the needed SCRs to be executed in the local map and only that, nothing more.

Reply:
Looking through the code it seems like indded on v2 we create the miniblocks using `ProcessMiniBlock` and the processing of the received block is done in `ProcessBlockTransactions`.

I've removedit from here